x-pack/plugins/cases/public/components/create/assignees.tsx

```diff
@@ -0,0 +1,148 @@
+  const { owner } = useCasesContext();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedOptions, setSelectedOptions] = useState<EuiComboBoxOptionOption[]>();
+  const { data: currentUserProfile, isLoading: isLoadingCurrentUserProfile } =
+    useGetCurrentUserProfile();
+
+  const { data: userProfiles, isLoading: isLoadingSuggest } = useSuggestUserProfiles({
+    name: searchTerm,
+    owners: owner,
+  });
+
+  const options =
+    userProfiles?.map((userProfile) => userProfileToComboBoxOption(userProfile)) ?? [];
+
+  const onSearchComboChange = (value: string) => {
+    if (!isEmpty(value)) {
+      setSearchTerm(value);
+    }
+  };
+
+  const isLoading = isLoadingForm || isLoadingCurrentUserProfile || isLoadingSuggest;
+
+  const renderOption = (
+    option: EuiComboBoxOptionOption,
+    searchValue: string,
+    contentClassName: string
+  ) => {
+    const { user, data, value } = option as EuiComboBoxOptionOption<string> & UserProfileWithAvatar;
+
+    return (
+      <EuiSelectableListItem
+        key={value}
+        prepend={<UserAvatar user={user} avatar={data.avatar} size="s" />}
+        className={contentClassName}
+        append={<EuiTextColor color="subdued">{user.email}</EuiTextColor>}
+      >
+        {getUserDisplayName(user)}
+      </EuiSelectableListItem>
+    );
+  };
+
+  return (
+    <UseField path="assignees" config={getConfig()}>
+      {(field) => {
+        const { setValue } = field;
+
+        const onComboChange = (currentOptions: EuiComboBoxOptionOption[]) => {
+          setSelectedOptions(currentOptions);
+          setValue(currentOptions.map((option) => comboBoxOptionToAssignee(option)));
+        };
+
+        const onSelfAssign = () => {
+          if (!currentUserProfile) {
+            return;
+          }
+
+          setSelectedOptions((prev) => [
+            ...(prev ?? []),
+            userProfileToComboBoxOption(currentUserProfile),
+          ]);
+          setValue([
+            ...(selectedOptions?.map((option) => comboBoxOptionToAssignee(option)) ?? []),
+            { uid: currentUserProfile.uid },
+          ]);
+        };
+
+        return (
+          <EuiFormRow
+            id="createCaseAssignees"
+            fullWidth
+            label={i18n.ASSIGNEES}
+            labelAppend={OptionalFieldLabel}
+            helpText={
+              <EuiLink data-test-subj="create-case-assign-yourself-link" onClick={onSelfAssign}>
+                {i18n.ASSIGN_YOURSELF}
+              </EuiLink>
+            }
+          >
+            <EuiComboBox
+              fullWidth
+              async
+              isLoading={isLoading}
+              options={options}
+              data-test-subj="createCaseAssigneesComboBox"
+              selectedOptions={selectedOptions}
+              isDisabled={isLoading}
+              onChange={onComboChange}
+              onSearchChange={onSearchComboChange}
```

Comment:
If you search and click on a user suggestion then the `onSearchChange` is called with an empty string. This will call the backend and return null making the suggestion list disappear. 


https://user-images.githubusercontent.com/7871006/188142014-8d7a8154-b53b-483a-a035-024102179bd9.mp4